app/assets/javascripts/components/Profile.es6.jsx

```diff
@@ -17,10 +20,78 @@ class Profile extends React.Component {
     this.renderDashboardList = this.renderDashboardList.bind(this);
     this.renderEditButton = this.renderEditButton.bind(this);
     this.renderUserContent = this.renderUserContent.bind(this);
+    this.loadMoreBooksOnScroll = this.loadMoreBooksOnScroll.bind(this);
+    this.loadMoreBooks = this.loadMoreBooks.bind(this);
+    this.loadBooksRequest = this.loadBooksRequest.bind(this);
+    this.renderBookLoader = this.renderBookLoader.bind(this);
+  }
+
+  componentDidMount(){
+    window.addEventListener('scroll', this.loadMoreBooksOnScroll);
+  }
+
+  renderBookLoader(){
+    if (this.state.request_more_books_sent) {
+      return(
+        <ul className="bookEntryList">
+          <hr/>
+          <div className="book-loader"></div>
+        </ul>
+      );
+    }
+  }
+
+  loadMoreBooksOnScroll(){
+    var scrollTop = (document.documentElement && document.documentElement.scrollTop) || document.body.scrollTop;
+    var scrollHeight = (document.documentElement && document.documentElement.scrollHeight) || document.body.scrollHeight;
+    var clientHeight = document.documentElement.clientHeight || window.innerHeight;
+    var scrolledToBottom = Math.ceil(scrollTop + clientHeight) >= scrollHeight;
+
+    if (scrolledToBottom) {
+      window.scrollTo(0, document.body.offsetHeight - 350);
+      this.loadMoreBooks();
+    }
+  }
+
+  loadMoreBooks(){
+    if (this.state.request_more_books_sent) {
+      return;
+    }
+
+    setTimeout(this.loadBooksRequest, 2000);
+
+    this.setState({request_more_books_sent: true});
+  }
+
+  loadBooksRequest(){
+    $.ajax({
+      url: "http://localhost:3000/books/show_more",
```

Comment:
this will need to be a relative path to work anywhere outside the local environment. `/books/show_more`